packages/fxa-auth-server/scripts/check-users.js

```diff
@@ -0,0 +1,225 @@
+// and if their password matches. It exports a CSV file that contains
+// whether the password matches, user has MFA enabled, secondary emails and
+// primary email verified.
+//
+// Example input file: ../test/scripts/fixtures/check_users.csv
+//
+// Usage: node scripts/check-users.js -i <input file> -o <output file>
+
+const fs = require('fs');
+const path = require('path');
+const program = require('commander');
+const pbkdf2 = require('../lib/crypto/pbkdf2');
+const hkdf = require('../lib/crypto/hkdf');
+
+program
+  .option('-d, --delimiter [delimiter]', 'Delimiter for input file', ':')
+  .option('-o, --output <filename>', 'Output filename to save results to')
+  .option(
+    '-i, --input <filename>',
+    'Input filename from which to read input if not specified on the command line'
+  )
+  .parse(process.argv);
+
+if (!program.input) {
+  console.error('input file must be specified');
+  process.exit(1);
+}
+
+const log = require('../lib/log')({});
+const config = require('../config').config.getProperties();
+const Token = require('../lib/tokens')(log, config);
+const AuthDB = require('../lib/db')(config, log, Token);
+const Password = require('../lib/crypto/password')(log, config);
+
+function sanitizeValue(v) {
+  if (v === undefined || v === null) {
+    return '';
+  }
+  return v;
+}
+class User {
+  constructor(email, password, db) {
+    this.email = email;
+    this.password = password;
+    this.db = db;
+  }
+
+  // FxA uses HKDF to derive the authPW from the user's password.
+  // This is typically done on the client side since FxA never sees the user's
+  // cleartext password.
+  async getCredentials(email, password) {
+    const stretch = await pbkdf2.derive(
+      Buffer.from(password),
+      hkdf.KWE('quickStretch', email),
+      1000,
+      32
+    );
+    this.authPW = await hkdf(stretch, 'authPW', null, 32);
+    this.unwrapBKey = await hkdf(stretch, 'unwrapBKey', null, 32);
+    return this;
+  }
+
+  async stats() {
+    try {
+      const accountRecord = await this.db.accountRecord(this.email);
+      const credentials = await this.getCredentials(
+        accountRecord.primaryEmail.normalizedEmail,
+        this.password
+      );
+
+      // Check the user password against the stored hash in DB
+      const password = new Password(
+        credentials.authPW,
+        accountRecord.authSalt,
+        accountRecord.verifierVersion
+      );
+      const verifyHash = await password.verifyHash();
+      const passwordMatch = await this.db.checkPassword(
+        accountRecord.uid,
+        verifyHash
+      );
+
+      // Check to see if user has MFA enabled
+      let mfaEnabled = false;
+      try {
+        const totpToken = await this.db.totpToken(accountRecord.uid);
+        if (totpToken) {
+          mfaEnabled = true;
+        }
+      } catch (err) {}
+
+      const s = {
+        email: this.email,
+        exists: true,
+        passwordMatch: passwordMatch.match,
```

Comment:
This is the only line changed, `passwordMatch` returns an object that has whether there was a match and which version is matched. Not sure if we care what version is matched but that a match exisits.